core/app/models/spree/tax/tax_helpers.rb

```diff
@@ -3,36 +3,14 @@ module Tax
     module TaxHelpers
       private
 
-      # Imagine with me this scenario:
-      # You are living in Spain and you have a store which ships to France.
-      # Spain is therefore your default tax rate.
-      # When you ship to Spain, you want the Spanish rate to apply.
-      # When you ship to France, you want the French rate to apply.
-      #
-      # Normally, Spree would notice that you have two potentially applicable
-      # tax rates for one particular item.
-      # When you ship to Spain, only the Spanish one will apply.
-      # When you ship to France, you'll see a Spanish refund AND a French tax.
-      # This little bit of code at the end stops the Spanish refund from appearing.
-      #
-      # For further discussion, see https://github.com/spree/spree/issues/4397 and https://github.com/spree/spree/issues/4327.
       def applicable_rates(order)
-        order_zone_tax_category_ids = rates_for_order(order).flat_map(&:tax_categories).map(&:id)
-        default_rates_with_unmatched_tax_category = rates_for_default_zone.to_a.delete_if do |default_rate|
-          (order_zone_tax_category_ids & default_rate.tax_categories.map(&:id)).any?
-        end
-
-        (rates_for_order(order) + default_rates_with_unmatched_tax_category).uniq
+        rates_for_order(order)
```

Comment:
Could we reduce the number of methods here or are these semi-public interfaces? In terms of `private`, but widely used by extensions/shops?

Reply:
We can reduce them. I chose to keep `applicable_rates`, because that's what's being used in calculators, and remove `rates_for_order`. Thank you!

Reply:
Thanks, but I think we can remove even more.

`applicable_rates` is never called outside of this helper. The calculators only use `rates_for_item`.

If we also move `sum_of_included_tax_rates` into `Spree::Calculator::DefaultTax` (the only place where this is called) we can reduce the module to:

```ruby
module Spree
  module Tax
    module TaxHelpers
      private

      def rates_for_item(item)
        rates = Spree::TaxRate.for_address(item.order.tax_address)
        rates.select do |rate|
          rate.tax_categories.map(&:id).include?(item.tax_category_id)
        end
      end
    end
  end
end
```

WDYT?

Reply:
And maybe we should rename the method to something meaningful then

`tax_rates_for(item)` maybe?

Reply:
I did not realize the module is only  called via `rates_for_item` by now, good catch! However, I did not change the name of the method. While it's not grammatically perfect, it conveys what the method does, and this way I don't have to change the calling code. 

I kept the ivar caching, but removed the two superfluous methods.

Reply:
Perfect! Thanks